cost/google/rightsize_vm_recommendations/google_rightsize_vm_recommendations.pt

```diff
@@ -0,0 +1,621 @@
+      for (var j = 0; j < ds_get_vms[i].vms.length; j++) {
+        if (param_zones.length === 0) {
+          collected_vms.push(ds_get_vms[i].vms[j])
+        } else {
+          for (var x = 0; x < param_zones.length; x++) {
+            if (ds_get_vms[i].vms[j].zone.indexOf(param_zones[x]) !== -1) {
+              collected_vms.push(ds_get_vms[i].vms[j])
+            }
+          }
+        }
+      }
+    }
+  EOS
+end
+
+#COMBINE VM DATA WITH VM CPU UTILIZATION DATA
+datasource "ds_vms_with_cpu_utilization" do
+  run_script $js_vms_with_cpu_utilization, $ds_collected_vms, $ds_calculated_utilization
+end
+
+script "js_vms_with_cpu_utilization", type: "javascript" do
+  result "vms_with_cpu_utilization"
+  parameters "ds_collected_vms", "ds_calculated_utilization"
+  code <<-EOS 
+    var vms_with_cpu_utilization = []
+    for (var i = 0; i < ds_calculated_utilization.length; i++) {
+      matching_vm = _.find(ds_collected_vms, function(v) { return v.id === ds_calculated_utilization[i].resourceID })
+      if (matching_vm !== undefined) {
+        // Rename some fields
+        matching_vm.resourceID = matching_vm.id
+        matching_vm.hostname = matching_vm.name || matching_vm.id
+        matching_vm.resourceType = matching_vm.kind
+
+        // Add CPU utilization data
+        matching_vm.resourceName = ds_calculated_utilization[i].resourceName
+        matching_vm.zone = ds_calculated_utilization[i].zone
+        matching_vm.cpuAverage = ds_calculated_utilization[i].cpuAverage
+        matching_vm.cpuMaximum = ds_calculated_utilization[i].cpuMaximum
+        matching_vm.cpuMinimum = ds_calculated_utilization[i].cpuMinimum
+        matching_vm.accountID = ds_calculated_utilization[i].accountID
+        matching_vm.accountName = ds_calculated_utilization[i].accountName
+        matching_vm.projectNumber = ds_calculated_utilization[i].projectNumber
+
+        vms_with_cpu_utilization.push(matching_vm)
+      }
+    }
+  EOS
+end
+
+#CREATE LIST OF RECOMMENDER DETAILS FOR RECOMMENDER API CALL USING COMBINED VM & CPU UTILIZATION DATA
+datasource "ds_recommenders" do
+  run_script $js_recommenders, $ds_vms_with_cpu_utilization
+end
+
+script "js_recommenders", type: "javascript" do
+  parameters "ds_vms_with_cpu_utilization"
+  result "results"
+  code <<-EOF
+    var results = []
+    for (var i = 0; i < ds_vms_with_cpu_utilization.length; i++) {
+      var result = {
+        projectNumber: ds_vms_with_cpu_utilization[i].projectNumber,
+        accountID: ds_vms_with_cpu_utilization[i].accountID,
+        region: ds_vms_with_cpu_utilization[i].zone,
+        requestNumber: results.length + 1,
+      }
+      var resultAtArray = _.find(results, function(r) { return r.projectNumber === result.projectNumber && r.accountID === result.accountID && r.region === result.region })
+      if (resultAtArray === undefined) {
+        results.push(result)
+      }
+    }
+  EOF
+end
+
+#CREATE LIST OF RECOMMENDATIONS
```

Comment:
Really minor change here to the comment - 
```suggestion
#GET LIST OF RECOMMENDATIONS
```

Reply:
@nia-vf1 , thank you, just changed this :)